lib/ansible/template/__init__.py

```diff
@@ -584,27 +608,14 @@ class Templar:
     '''
 
     def __init__(self, loader, shared_loader_obj=None, variables=None):
-        variables = {} if variables is None else variables
-
+        # NOTE shared_loader_obj is deprecated, ansible.plugins.loader is used
```

Comment:
Do we have a deprecation warning somewhere for 3rd party stuff or has it not been usable for a while? It looks like we still have code that's passing shared_loader_obj to Templar too. Looks good to me besides that.

Reply:
Looks like I missed some places where we still pass `shared_loader_obj` to `Templar` which I have just removed. `shared_loader_obj` variables that we still have in the codebase are basically `ansible.plugins.loader` now.

I am not aware of any deprecation warning for 3rd party stuff nor do I know how do we handle cases like this. I don't think we need to necessarily keep that arg there since we don't support Python API. Do you think we should remove it?

Reply:
I was mostly just curious since you have `"used"` on the line below and it made me think it might not have really worked. It's not a direct comparison, but we used to allow importing cache plugins directly so when I switched that over to use the config manager I kept backwards compat and deprecated the behavior to try to give external stuff a heads up (like here https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/cache/__init__.py#L65). It's probably less important in the current landscape of collections.